src/target/riscv/riscv.c

```diff
@@ -3061,30 +3061,22 @@ static int riscv_virt2phys(struct target *target, target_addr_t virtual, target_
 			virtual, physical);
 }
 
-static int riscv_read_phys_memory(struct target *target, target_addr_t phys_address,
-			uint32_t size, uint32_t count, uint8_t *buffer)
+static int check_virt_memory_access(struct target *target, target_addr_t address,
+			uint32_t size, uint32_t count)
 {
-	RISCV_INFO(r);
-	return r->read_memory(target, phys_address, size, count, buffer, size);
+	bool crosses_page_boundary = RISCV_PGBASE(address + size * count - 1) != RISCV_PGBASE(address);
+	if (address % size && crosses_page_boundary) {
```

Comment:
Detail for readability:

```suggestion
	const bool is_misaligned = address % size != 0;
	const bool crosses_page_boundary = RISCV_PGBASE(address + size * count - 1) != RISCV_PGBASE(address);
	if (is_misaligned && crosses_page_boundary) {
```

Reply:
Thanks. Addressed.